x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -155,6 +151,34 @@ export const useFetchIndex = (
             includeUnmapped
           );
 
+          const fields = dv.fields.map((field) => field.toSpec());
+          if (includeUnmapped) {
+            try {
+              const fieldNameConflictDescriptionsMap = (await data.dataViews
+                .getFieldsForIndexPattern(dv, {
+                  pattern: '',
+                  includeUnmapped: true,
+                })
+                .then((fieldsToReduce) => {
+                  return fieldsToReduce.reduce(
+                    (acc, f) => ({ ...acc, [f.name]: f.conflictDescriptions }),
+                    {} as { [x: string]: Record<string, string[]> | undefined }
+                  );
+                })) as { [x: string]: Record<string, string[]> | undefined };
+
+              dv.fields.replaceAll([
+                ...fields.map((field) => ({
+                  ...field,
+                  ...(fieldNameConflictDescriptionsMap[field.name] != null
+                    ? { conflictDescriptions: fieldNameConflictDescriptionsMap[field.name] }
+                    : {}),
+                })),
+              ]);
```

Comment:
This is my poor-performing code - sorry I offered it up as a working solution from my other PR.

Anyways, I feel we can reduce this to a single iteration with the below code and it should work:


```suggestion
             const fieldNameConflictDescriptionsMap =
                await data.dataViews.getFieldsForIndexPattern(dv, {
                  pattern: '',
                  includeUnmapped: true,
                });
              fieldNameConflictDescriptionsMap.forEach((field) => {
                dv.fields.update({ ...field });
              });
```